### ISSUE TITLE
Doesn't work since today morning

### ISSUE THREAD

**Issue by karoljas96** [turnermm/ckgedit#541]
This morning dokuwiki.org had a server crash. From this moment ckgedit does not work. When trying to edit a page, an "Error" alert appears. In addition, the source of the page is:
 Uncaught ReferenceError: DOKU_BASE is not defined
    at script-cmpr.js:1
    at script-cmpr.js:1

**Comment by turnermm** [comment]
What happens at dokuwiki.org has nothing to do with ckgedit.  That is a javascript error that could originate from anywhere.  I just reinstalled ckgedit and found no issues.  I see that the dokuwiki language tool has been active today. What is your language?  Perhaps someone made an error updating a language file.

**Comment by karoljas96** [comment]
Thank you for your quick response. I use Polish (pl). I reinstalled the plugin and touched cache but no effect. 
This is the line where DOKU_BASE is not defined: 
`var HTMLParser;var HTMLParserInstalled=true;var HTMLParser_Elements=new Array();(function(){var k=/^<(\w+)((?:\s+\w+(?:\s*=\s*(?:(?:"[^"]*")|(?:'[^']*')|[^>\s]+))?)*)\s*(\/?)>/,c=/^<\/(\w+)[^>]*>/,f=/(\w+)(?:\s*=\s*(?:(?:"((?:\\.|[^"])*)")|(?:'((?:\\.|[^'])*)')|([^>\s]+)))?/g;var e=b("br,col,hr,img");var a=b("blockquote,center,del,div,dl,dt,hr,iframe,ins,li,ol,p,pre,table,tbody,td,tfoot,th,thead,tr,ul");var h=b("a,abbr,acronym,b,big,br,cite,code,del,em,font,h1,h2,h3,h4,h5,h6,i,img,ins,kbd,q,s,samp,small,span,strike,strong,sub,sup,tt,u,var");var d=b("colgroup,dd,dt,li,options,p,td,tfoot,th,thead,tr");var j=b("checked,disabled,ismap,noresize,nowrap,readonly,selected");var g=b("script,style");var l="http://"+location.host+DOKU_BASE+"/lib/plugins/ckgedit/fckeditor/userfiles/blink.jpg?nolink&33x34";HTMLParser=this.HTMLParser=function(n,w){var q,r,o,t=[],u=n;n=n.replace(/(<img.*?src="data:image\/\w+;base64,\s*)(.*?)(\/>)/gm,function(x,A,z){var y=false;if(A.match(/msword/)){y=true;x=x.replace(/msword/,"")}if(z.length>2500000&&!y){jQuery("#dw__editform").append('<input type="hidden" id="broken_image" name="broken_image" value="'+z.length+'" />');return"{{"+l+"}}"}return x});n=n.replace(/~~OPEN_HTML_BLOCK~~/gm,"~~START_HTML_BLOCK~~");n=n.replace(/~~END_HTML_BLOCK~~/gm,"~~CLOSE_HTML_BLOCK~~");if(n.match(/~~START_HTML_BLOCK~~/gm)){if(!JSINFO.htmlok){n=n.replace(/~~START_HTML_BLOCK~~|~~CLOSE_HTML_BLOCK~~/gm,"")}n=n.replace(/(<p>)*\s*~~START_HTML_BLOCK~~\s*(<\/p>)*([\s\S]+)~~CLOSE_HTML_BLOCK~~\s*(<\/p>)*/gm,function(x,z,B,A,y){A=A.replace(/<\/?div.*?>/gm,"");A=A.replace(/<code>/gm,"");A=A.replace(/<\/code>/gm,"");A=A.replace(/</gm,"&lt;");A=A.replace(/<\//gm,"&gt;");return"~~START_HTML_BLOCK~~\n\n"+A+"\n\n~~CLOSE_HTML_BLOCK~~\n\n"})}n=n.replace(/(<sup\s+class=\"dwfcknote fckgL\d+\"\>fckgL\d+\s*\<\/sup\>)\<\/a\>/gm,function(z,y,x){return("</a>"+y)});var s=n.indexOf("data-");if(s!=-1){n=n.replace(/(<\w+)([^>]+)>/gm,function(y,x,z){z=z.replace(/data-[\w\-]+\s*=\s*(\"|\')\w+(\"|\')/,"");return x+z+">"})}t.last=function(){return this[this.length-1]};while(n){r=true;if(!t.last()||!g[t.last()]){if(n.indexOf("<!--")==0){q=n.indexOf("-->");if(q>=0){if(w.comment){w.comment(n.substring(4,q))}n=n.substring(q+3);r=false}}else{if(n.indexOf("</")==0){o=n.match(c);if(o){n=n.substring(o[0].length);o[0].replace(c,p);r=false}}else{if(n.indexOf("<")==0){o=n.match(k);if(o){n=n.substring(o[0].length);o[0].replace(k,m);r=false}}}}if(r){q=n.indexOf("<");var v=q<0?n:n.substring(0,q);n=q<0?"":n.substring(q);if(w.chars){w.chars(v)}}}else{n=n.replace(new RegExp("(.*)</"+t.last()+"[^>]*>"),function(x,y){y=y.replace(/<!--(.*?)-->/g,"$1").replace(/<!\[CDATA\[(.*?)]]>/g,"$1");if(w.chars){w.chars(y)}return""});p("",t.last())}if(n==u){throw"Parse Error: "+n}u=n}p();function m(x,A,B,y){if(a[A]){while(t.last()&&h[t.last()]){p("",t.last())}}if(d[A]&&t.last()==A){p("",A)}y=e[A]||!!y;if(!y){t.push(A)}if(w.start){var z=[];B.replace(f,function(D,C){var E=arguments[2]?arguments[2]:arguments[3]?arguments[3]:arguments[4]?arguments[4]:j[C]?C:"";z.push({name:C,value:E,escaped:E.replace(/(^|[^\\])"/g,'$1\\"')})});if(w.start){w.start(A,z,y)}}}function p(x,z){if(!z){var A=0}else{for(var A=t.length-1;A>=0;A--){if(t[A]==z){break}}}if(A>=0){for(var y=t.length-1;y>=A;y--){if(w.end){w.end(t[y])}}t.length=A}}};function b(p){var o={},m=p.split(",");for(var n=0;n<m.length;n++){o[m[n]]=true}return o}})();function HTMLParser_test_result(a){var d="";for(i=0;i<a.length;i++){var c=a.charAt(i);if(a.charCodeAt(i)==10){c="\\n"}if(a.charCodeAt(i)==32){c="SP"}var b=c+" ";d+=b;if(a.charCodeAt(i)==10){d+="\n"}}if(!confirm(d)){return false}return true}function hide_backup_msg(){document.getElementById("backup_msg").style.display="none";return false}function show_backup_msg(a){document.getElementById("backup_msg").style.display="block";document.getElementById("backup_msg_area").innerHTML="Backed up to: "+a;return false}function remove_draft(){}function dwedit_draft_delete(){}function setEdHeight(a){a=parseInt(a);document.cookie="ckgEdht="+a+';expires="";path='+JSINFO.doku_base}function ckgd_setImgPaste(b){var a=JSINFO.ckgEdPaste?JSINFO.ckgEdPaste:"";if(a=="on"){b="off"}else{b="on"}JSINFO.ckgEdPaste=b;document.cookie="ckgEdPaste="+b+';expires="Thu, 18 Dec 2575 12:00:00 UTC";path='+JSINFO.doku_base;alert(LANG.plugins.ckgedit.ckg_paste_restart+" "+LANG.plugins.ckgedit[b])}function ckg_RawImgMsg(){return LANG.plugins.ckgedit.broken_image_1+"\n"+LANG.plugins.ckgedit.broken_image_2}function GetE(a){return document.getElementById(a)}var dokuBase=location.host+DOKU_BASE;if(window.getSelection!=undefined){var doku_ckg_getSelection=window.getSelection;window.getSelection=function(a){if(!a){a=GetE("wiki__text")}return doku_ckg_getSelection(a)}}function ckgedit_seteditor_priority(a,c,b){var e={Y:"Dokuwiki",N:"CKEditor"};if(typeof a==="undefined"){if(b[0].checked){a=b[0].value}else{if(b[1].checked){a=b[1].value}}}var d="dw_val="+a;d+="&call=cked_selector";d+="&dwp_client="+c;jQuery.post(DOKU_BASE+"lib/exe/ajax.php",d,function(f){if(f=="done"){if(!a){alert(LANG.plugins.ckgedit.dwp_not_sel)}else{alert(LANG.plugins.ckgedit.dwp_updated+e[a])}}else{alert(LANG.plugins.ckgedit.dwp_save_err+f)}},"html")}function ckged_get_unlink_size(b){var a="call=cked_deletedsize";a+="&cked_delid="+b;jQuery.post(DOKU_BASE+"lib/exe/ajax.php",a,function(c){if(c){JSINFO.ckg_del_sz=c}else{}},"html")}function ckged_setmedia(d,a,b){var c="call=cked_upload";c+="&ckedupl_id="+d;if(a){c+="&ckedupl_del=D&delsize="+JSINFO.ckg_del_sz}jQuery.post(DOKU_BASE+"lib/exe/ajax.php",c,function(e){if(e){if(b){b.postMessage(JSINFO.doku_url,JSINFO.doku_url)}}else{}},"html")}jQuery(document).ready(function(){if(JSINFO.hide_captcha_error=="hide"){jQuery("div.error").hide()}});jQuery(document).ready(function(){jQuery("#editor_height").keydown(function(a){if(a.which==13){a.preventDefault()}});$dokuWiki=jQuery(".dokuwiki");jQuery(".editbutton_table button").click(function(){var a=this.form;jQuery("<input />").attr("type","hidden").attr("name","mode").attr("value","dwiki").appendTo(jQuery(a));jQuery("<input />").attr("type","hidden").attr("name","fck_preview_mode").attr("value","nil").appendTo(jQuery(a))});if(typeof(JSINFO.dbl_click_auth!=="undefined")&&JSINFO.dbl_click_auth==""){return}if(!JSINFO.ckg_dbl_click){return}if(jQuery(".editbutton_section",$dokuWiki).length>0){jQuery('[class^="sectionedit"], div[class^="level"]',$dokuWiki).dblclick(function(){var a=jQuery(this).nextAll(".editbutton_section:eq(0)").children("form:eq(0)");jQuery("<input />").attr("type","hidden").attr("name","mode").attr("value","dwiki").appendTo(jQuery(a));jQuery("<input />").attr("type","hidden").attr("name","fck_preview_mode").attr("value","nil").appendTo(jQuery(a));a.submit()})}if(JSINFO.ckg_template.match(/bootstrap/)&&jQuery("div.editButtons").length>0){jQuery("div.editButtons input").each(function(a){if(jQuery(this).hasClass("btn-success")){jQuery(this).removeClass("btn-success")}if(jQuery(this).hasClass("btn-danger")){jQuery(this).removeClass("btn-danger")}})}});function ckg_edit_mediaman_insert(h,c,a,f){var k,d,n,j;var m=a.substring(1).split("&");k="detail";for(var g in m){var b=m[g];if(b.match(/^\d+$/)){d=b}else{if(b.match(/^\w+$/)){k=b}}}switch(f){case"2":j="medialeft";break;case"3":j="mediacenter";break;case"4":j="mediaright";break;default:j="";break}var e=CKEDITOR.instances.wiki__text._.filebrowserFn;var l=DOKU_BASE+"lib/exe/fetch.php?media="+c;CKEDITOR.tools.callFunction(e,l,function(){var o=this.getDialog();if(o.getName()=="image"){if(j!=null){o.getContentElement("info","cmbAlign").setValue(j)}if(k!=null){o.getContentElement("info","cmbLinkType").setValue(k)}if(d!=null){o.getContentElement("info","txtWidth").setValue(d);o.dontResetSize=true}}})}function ckg_edit_mediaman_insertlink(a,e,c,b){var d=CKEDITOR.instances.wiki__text._.filebrowserFn;CKEDITOR.tools.callFunction(d,e,function(){var f=this.getDialog();if(f.getName()=="link"){f.getContentElement("info","media").setValue(e)}})}function getCookie(a){var b=new RegExp(a+"=([^;]+)");var c=b.exec(document.cookie);return(c!=null)?decodeURIComponent(c[1]):null};`

**Comment by turnermm** [comment]
Whenever there is a javascript error DOKU_BASE is almost always undefined, and this is because an error somewhere prior to DOKU_BASE has occurred, not because it is in fact undefined.  Have you installed any other plugins?

It wouldn't likely be the Polish language file, since it was last updated in January, and moreover it doesn't emit any errors on testing.

**Comment by karoljas96** [comment]
Yeah I apologise, I use default (en_US) ckgedit language. I turned off all other plugins and it doesn't help but finally I found a solution. I deleted userscript.js file from dokuwiki/conf. This file was totaly empty so I have no idea why it caused an error. But everytime I create this file again and refresh cache fatal error with ckgedit apears. Strange, isn't it? 
Anyway thank you so much for your help Dev :)
